service-base/src/main/java/org/eclipse/hono/service/amqp/RequestResponseEndpoint.java

```diff
@@ -367,12 +366,10 @@ public final void onLinkAttach(final ProtonConnection con, final ProtonSender se
             logger.debug("client [{}] closed sender link, removing associated event bus consumer [{}]",
                     sender.getName(), replyConsumer.address());
 
-            deallocateReceiverForReplyTo(replyToAddress.toString());
+            deallocateReceiverForReplyTo(replyTo);
             unregisterConsumerForConnection(con, replyTo, replyConsumer);
 
-            if (senderClosed.succeeded()) {
-                senderClosed.result().close();
-            }
+            senderClosed.result().close();
```

Comment:
IMHO this will yield a `NullPointerException` if the peer included an error in its detach frame. You will need to use `sender.close()` instead ...

Reply:
🤦 … True, I completely overlooked that.